echopype/convert/api.py

```diff
@@ -427,28 +429,43 @@ def open_raw(
         sonar_model=sonar_model,
         params=_set_convert_params(convert_params),
     )
-    # Set up echodata object
-    echodata = EchoData(source_file=file_chk, xml_path=xml_chk, sonar_model=sonar_model)
+
+    # Setup tree dictionary
+    tree_dict = {}
     # Top-level date_created varies depending on sonar model
     if sonar_model in ["EK60", "ES70", "EK80", "ES80", "EA640"]:
-        echodata.top = setgrouper.set_toplevel(
-            sonar_model=sonar_model, date_created=parser.config_datagram["timestamp"]
+        tree_dict["Top-level"] = setgrouper.set_toplevel(
+            sonar_model=sonar_model,
+            date_created=parser.config_datagram["timestamp"],
         )
     else:
-        echodata.top = setgrouper.set_toplevel(
+        tree_dict["Top-level"] = setgrouper.set_toplevel(
             sonar_model=sonar_model, date_created=parser.ping_time[0]
         )
-    echodata.environment = setgrouper.set_env()
-    echodata.platform = setgrouper.set_platform()
+    tree_dict["Environment"] = setgrouper.set_env()
+    tree_dict["Platform"] = setgrouper.set_platform()
     if sonar_model in ["EK60", "ES70", "EK80", "ES80", "EA640"]:
-        echodata.nmea = setgrouper.set_nmea()
-    echodata.provenance = setgrouper.set_provenance()
-    echodata.sonar = setgrouper.set_sonar()
-    # Beam_group2 group only exist if EK80 has both complex and power/angle data
-    if sonar_model in ["EK80", "ES80", "EA640"]:
-        echodata.beam, echodata.beam_power = setgrouper.set_beam()
-    else:
-        echodata.beam = setgrouper.set_beam()
-    echodata.vendor = setgrouper.set_vendor()
+        tree_dict["Platform/NMEA"] = setgrouper.set_nmea()
+    tree_dict["Provenance"] = setgrouper.set_provenance()
+    tree_dict["Sonar"] = setgrouper.set_sonar()
+
+    # Set multi beam groups
+    beam_groups = setgrouper.set_beam()
+    if isinstance(beam_groups, xr.Dataset):
+        # if it's a single dataset like the ek60,
+        # make into list
+        beam_groups = [beam_groups]
```

Comment:
@emiliom Do you think there's a way that `set_beam()` always return a list of `xr.Dataset` even if there's only one?

Reply:
@lsetiawan I've looked at `set_groups_ek80.py`, which already returns a list of `xr.Dataset`, vs ek60 and azfp which return a single `xr.Dataset`. I don't see any problem with changing `set_beam()` so it always returns a list. I like that suggestion. Though it's not necessarily for this PR, right?

Reply:
> I'll keep this check for now, but if you are planning to change this

Nope, no plan to do this. I don't want to expand the scope of this PR too much.